Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/BaseInteractiveElement.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    public abstract class BaseInteractiveElement :
+        MonoBehaviour,
+        IMixedRealityFocusHandler
+    {
```

Comment:
Here is the proposed structure of integrating button state setting logic, the elements in grey have not been added. 
This structure will add far interaction support to the current Pressable Button
![image](https://user-images.githubusercontent.com/53493796/91915303-0bb29780-ec6f-11ea-8dab-e138aafeac3e.png)